Sources/ArgumentParser/Parsable Properties/CompletionKind.swift

```diff
@@ -9,56 +9,225 @@
+    /// - either:
+    ///   - all the arguments on the command line (bash & zsh, presumably fish
+    ///     4+)
+    ///   - the arguments on the command line only up to & including the
+    ///     argument for which completions are being requested (fish 3-)
+    ///
+    /// The arguments are passed to Swift verbatim, not unquoted. e.g., the
+    /// representation in Swift of the shell argument `"abc\\""def"` would be
+    /// exactly the same, including the quotes & the double backslash.
+    ///
+    /// Due to limitations in fish 3, arguments are passed to Swift unquoted.
+    /// e.g., the aforementioned example argument would be passed as `abc\def`.
+    /// fish 4 will not be constrained by such limitations, so this behavior
+    /// will be able to be fixed in a subsequent update to Swift Argument
+    /// Parser.
     case custom(@Sendable ([String]) -> [String])
   }
 
   internal var kind: Kind
 
-  /// Use the default completion kind for the value's type.
+  /// Use the default completion kind for the argument's or option value's type.
   public static var `default`: CompletionKind {
     CompletionKind(kind: .default)
   }
 
-  /// Use the specified list of completion strings.
+  /// The completion candidates are the elements of `words`.
+  ///
+  /// Completion candidates are interpreted by the requesting shell as literals.
+  /// They must be neither escaped nor quoted; Swift Argument Parser escapes or
+  /// quotes them as necessary for the requesting shell.
+  ///
+  /// The completion candidates are included in a completion script when it is
+  /// generated.
   public static func list(_ words: [String]) -> CompletionKind {
     CompletionKind(kind: .list(words))
   }
 
-  /// Complete file names.
+  // swift-format-ignore: BeginDocumentationCommentWithOneLineSummary
+  /// The completion candidates include:
+  ///
+  /// - all directory names
+  /// - if `extensions` is empty, all file names
+  /// - if `extensions` is not empty, all file names whose respective
+  ///   extension matches any element of `extensions`
+  ///
+  /// Given file extensions must not include the `.` initial extension
+  /// separator.
+  ///
+  /// Given file extensions are parsed by the requesting shell as globs; Swift
+  /// Argument Parser does not perform any escaping or quoting.
+  ///
+  /// In zsh, `EXTENDED_GLOB` & `NULL_GLOB` are set, while `KSH_GLOB` & `SH_GLOB` are unset
+  /// ? `BARE_GLOB_QUAL`, `GLOB_DOTS`, `GLOB_STAR_SHORT`, `GLOB_SUBST`, `NOMATCH`, `NUMERIC_GLOB_SORT`
+  ///
+  /// The directory/file filter & the given list of extensions are included in a
+  /// completion script when it is generated.
   public static func file(extensions: [String] = []) -> CompletionKind {
     CompletionKind(kind: .file(extensions: extensions))
   }
 
-  /// Complete directory names.
+  /// The completion candidates are directory names.
+  ///
+  /// The directory filter is included in a completion script when it is
+  /// generated.
```

Comment:
I like these cues about how/where the different completion configurations are used 👍🏻